shardmanager.go

```diff
@@ -90,67 +90,86 @@ func (n *ShardManager) connectShardNodes(ctx context.Context, shardID ShardIDTyp
 	}
 
 	// borrowed from `bootstrapConnect`, should be modified/refactored and tested
-	errs := make(chan error, len(pinfos))
 	var wg sync.WaitGroup
 	for _, p := range pinfos {
 		wg.Add(1)
-		go func(p pstore.PeerInfo) {
+		go func(ctx context.Context, p pstore.PeerInfo) {
```

Comment:
It seems fine if we directly reference `ctx` in the goroutine, without passing it explicitly as an argument to this goroutine function, thanks to the _Closure_?